service/app/controllers/usermanagement/UserController.java

```diff
@@ -127,13 +127,16 @@
 
   private Promise<Result> handleGetUserProfile(String operation, String userId) {
     final String requestedFields = request().getQueryString(JsonKey.FIELDS);
-
+    final String provider = request().getQueryString(JsonKey.PROVIDER);
```

Comment:
code can be generic - so next time key is added no need to handle any code- all request parameters go into context params.